File: backend/api/src/routes/connections.ts
import express, { Router } from "express";
import deleteconnections from "../controllers/connections/deleteConnections";
import getConnections from "../controllers/connections/getConnections";
import patchConnections from "../controllers/connections/patchConnections";
import postConnections from "../controllers/connections/postConnections";
import connectionsErrorHandler from "../middlewares/ErrorHandlers/connectionsErrorHandler";
const router: Router = express.Router();

router.use(express.json());

router.get("/", getConnections);
router.post("/", postConnections);
router.patch("/", patchConnections);
router.delete("/", deleteconnections);

router.use(connectionsErrorHandler);

export default router;

File: backend/internal/routes/connections.go
package routes

import (
	"net/http"
	"os"
	"path"
	"path/filepath"

	. "github.com/leminhnguyenai/notion-calendar/backend/internal/controllers/connectionsControllers"
	"github.com/leminhnguyenai/notion-calendar/backend/internal/middlewares"
	"github.com/leminhnguyenai/notion-calendar/backend/internal/services/api"
)

func Connections() *api.Router {
	absPath, err := os.Executable()
	if err != nil {
		panic(err)
	}

	dirname := filepath.Dir(absPath)

	router := api.NewRouter()

	validateUserInputConn := middlewares.InitSchemaValidation(
		path.Join(dirname, ".././internal/schemas/userInputNotionConn.json"),
	)

	validateDeleteConn := middlewares.InitSchemaValidation(
		path.Join(dirname, ".././internal/schemas/deleteNotionConn.json"),
	)

	router.Use(middlewares.ValidateToken)

	router.GET("/", http.HandlerFunc(GetConnections))
	router.POST("/", validateUserInputConn(http.HandlerFunc(PostConnection)))
	router.PATCH("/", validateUserInputConn(http.HandlerFunc(PatchConnection)))
	router.DELETE("/", validateDeleteConn(http.HandlerFunc(DeleteConnection)))

	return router
}
